packages/framer-motion/src/animation/animators/js/index.ts

```diff
@@ -297,12 +308,21 @@ export function animateValue<V = number>({
             newTime = secondsToMilliseconds(newTime)
 
             time = newTime
-            if (holdTime !== null || !animationDriver) {
+            if (holdTime !== null || !animationDriver || speed === 0) {
                 holdTime = newTime
             } else {
-                startTime = animationDriver.now() - newTime
+                startTime = animationDriver.now() - newTime / speed
             }
         },
+        get speed() {
+            return speed
+        },
+        set speed(newSpeed: number) {
+            if (newSpeed === speed || !animationDriver) return
+            speed = newSpeed
+            startTime = animationDriver.now() - time / speed
```

Comment:
Why not use `set time()` here?

Reply:
We also should check for speed `0` here, otherwise we'll get an error.

Reply:
Ah yeah great point also that function does check for `0`